src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -64,10 +64,15 @@ object ScalafixPlugin extends AutoPlugin {
         "Optional list of artifacts to resolve to run custom rules. " +
           "Can be set in ThisBuild or at project-level."
       )
+    @deprecated(
+      "scalafixScalaBinaryVersion now follows scalaBinaryVersion by default",
+      "0.12.1"
+    )
     val scalafixScalaBinaryVersion: SettingKey[String] =
       settingKey[String](
-        "The Scala binary version used for scalafix execution. Can be set in ThisBuild or at project-level. " +
-          "Custom rules must be compiled against that binary version. Defaults to 2.12."
+        "The Scala binary version used for scalafix execution. Must be set at project-level. " +
```

Comment:
Usage of `derive()` prevents visibility of user-defined `ThisBuild / scalafixScalaBinaryVersion` (which was the only option until [recently](https://github.com/scalacenter/sbt-scalafix/pull/380)) at project-level. That said, I can't think of a use case where the user would need to force a version which does not match scalaBinaryVersion (thus the deprecation).